src/ds3.c

```diff
@@ -2524,6 +2455,83 @@ ds3_error* ds3_get_available_chunks(const ds3_client* client, const ds3_request*
     return NULL;
 }
 
+static ds3_error* _parse_jobs_list(const ds3_log* log, xmlDocPtr doc, ds3_get_jobs_response** _response){
+    ds3_get_jobs_response* response;
+    xmlNodePtr root, child_node;
+    GPtrArray* jobs_array = g_ptr_array_new();
+    ds3_bulk_response* job;
+    ds3_error* error;
+
+    root = xmlDocGetRootElement(doc);
+
+    if(element_equal(root, "Jobs") == false) {
+        char* message = g_strconcat("Expected the root element to be 'Jobs'.  The actual response is: ", root->name, NULL);
+        xmlFreeDoc(doc);
+        ds3_error* error = _ds3_create_error(DS3_ERROR_INVALID_XML, message);
+        g_free(message);
+        return error;
+    }
+
+    response = g_new0(ds3_get_jobs_response, 1);
+
+    for(child_node = root->xmlChildrenNode; child_node != NULL; child_node = child_node->next) {
+        if(element_equal(child_node, "Job") == true) {
+            job = g_new0(ds3_bulk_response, 1);
+            error = _parse_bulk_response_attributes(log, doc, child_node, job);
+            if( error )
+            {
+              LOG(log, DS3_ERROR, "Error parsing bulk_response element");
+            }
+            g_ptr_array_add(jobs_array, job);
+        } else{
+            // Invalid XML block
+            LOG(log, DS3_ERROR, "Unknown child node: (%s)", child_node->name);
+        }
+    }
+
+    response->jobs_size = jobs_array->len;
+    response->jobs = (ds3_bulk_response**) jobs_array->pdata;
+    g_ptr_array_free(jobs_array, FALSE);
+
+    *_response = response;
+    return NULL;
+}
+
+ds3_error* ds3_get_jobs(const ds3_client* client, const ds3_request* request, ds3_get_jobs_response** response) {
+    ds3_error* error;
+    GByteArray* xml_blob = g_byte_array_new();
+    ds3_get_jobs_response* get_jobs_response = NULL;
+    GHashTable* response_headers = NULL;
+    xmlDocPtr doc;
+
+    error = _net_process_request(client, request, xml_blob, load_buffer, NULL, NULL, &response_headers);
+
+    if (error != NULL) {
+        if (response_headers != NULL) {
```

Comment:
Looking at the code for glib.  If `response_headers` is `NULL` it will return without an error.  So you should not have to check to make sure that it not null.